src/lib.rs

```diff
@@ -1609,12 +1609,33 @@ pub unsafe trait TryFromBytes {
     /// there exist types whose bit validity is ambiguous. See
     /// [here][TryFromBytes#what-is-a-valid-instance] for a discussion of how
     /// these cases are handled.
+    ///
+    /// # Compile-Time Assertions
+    ///
+    /// This method cannot yet be used on unsized types whose dynamically-sized
```

Comment:
```suggestion
    /// This method cannot be used on unsized types whose dynamically-sized
```

IMO we shouldn't imply that we'll lift this restriction in the future. We might, but it's not guaranteed, and it's also pointless cognitive burden for users to think about our long-term plans when there's no specific timeline and so there's no meaningful difference between "you can't yet" and "you can't."

Same elsewhere in this PR.

Reply:
We don't want folks relying on this function as a compile-time assert on DST ZSTiness so the "yet" is load-bearing.